extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/MySQLDialectConfig.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.hibernate.orm.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocDefault;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
+import io.smallrye.config.WithConverter;
+
+/**
+ * Configuration specific to the Hibernate ORM {@linkplain org.hibernate.dialect.MySQLDialect},
+ * though may also affect other dialects such as {@linkplain org.hibernate.dialect.MariaDBDialect}.
+ *
+ * @author Steve Ebersole
+ */
+@ConfigGroup
+public interface MySQLDialectConfig {
+    /**
+     * Specifies the bytes per character to use based on the database's configured
+     * <a href="https://dev.mysql.com/doc/refman/8.0/en/charset-charsets.html">charset</a>.
+     *
+     * @see org.hibernate.cfg.DialectSpecificSettings#MYSQL_BYTES_PER_CHARACTER
```

Comment:
Note:

1. Such links won't appear in generated documentation.
2. Application users are unlikely to reach this javadoc directly, they'll most likely read the rendered version included in the Quarkus documentation -- and in IDE plugins.

That's why we tend to do things like this: https://github.com/quarkusio/quarkus/blob/eeaecac7573dced42fa73c067ec84602c75120e3/extensions/hibernate-search-orm-outbox-polling/runtime/src/main/java/io/quarkus/hibernate/search/orm/outboxpolling/runtime/HibernateSearchOutboxPollingBuildTimeConfigPersistenceUnit.java#L88-L90
Reminder that Asciidoc syntax requires `@asciidoctlet` in the Javadoc comment.

`hibernate-orm-docs-url` is defined [here](https://github.com/quarkusio/quarkus/blob/b97f08b1d8642f184095cb704bc08eb0d96a6c14/docs/src/main/asciidoc/_attributes.adoc#L54), in case you want to add links to Hibernate ORM Javadocs -- but since nowadays you have a list of documentation properties in the Hibernate ORM user guide, I guess it could be more practical to link directly there?

Reply:
> Reminder that Asciidoc syntax requires @asciidoctlet in the Javadoc comment.

+1, sure just have not been using asciidoc formatting yet.  But clearly I will need to start :)

Reply:
> but since nowadays you have a list of documentation properties in the Hibernate ORM user guide, I guess it could be more practical to link directly there?

I think that's a good idea, Currently we don't generate url anchors for each setting as we'd need to, but https://hibernate.atlassian.net/browse/HHH-18654